corehq/apps/accounting/forms.py

```diff
@@ -566,7 +566,7 @@ def clean(self):
             and start_date > self.cleaned_data['end_date']):
             raise ValidationError("End date must be after start date.")
 
-        if self.cleaned_data['end_date'] <= datetime.date.today():
+        if self.cleaned_data['end_date'] and self.cleaned_data['end_date'] <= datetime.date.today():
```

Comment:
looked into this a bit more. the `end_date` field is a `DateField` which makes it a `datetime.date` and has no timezone. We are comparing it to a timezone aware `datetime.datetime`. I think the proper way to do this is to change the `DateField` to DateTimeField`. That said, I don't think it's worth it at the moment. feel free to push back if you think it's something we should change soon